Components/Modal/modal.js

```diff
@@ -0,0 +1,25 @@
+const modalToggleBtn = document.querySelector(".modal-btn");
+const modal = document.querySelector(".modal");
+const closeBtn = document.querySelector(".btn-close");
+const modalOverlayBg = document.querySelector(".modal-overlay");
+
+modalToggleBtn.addEventListener("click", () => {
+    if(!modal.classList.contains("show")) {
+        modal.classList.add("show");
+        document.body.style.overflow = "hidden";
+        modalOverlayBg.style.display = "block";
+        modalOverlayBg.style.backgroundColor = "var(--modal-overlay-color)";
+    } else {
+        modal.classList.remove("show");
+        document.body.style.overflow = "auto";
+        modalOverlayBg.style.display = "block";
+        modalOverlayBg.style.backgroundColor = "var(--white-color)";
+    }
+})
+
+closeBtn.addEventListener("click", () => {
+    modal.classList.remove("show");
+    document.body.style.overflow = "auto";
+    modalOverlayBg.style.display = "none";
+    modalOverlayBg.style.backgroundColor = "var(--white-color)";
```

Comment:
I think these configurations would be better off as a class being added rather than manually changing it here.

Reply:
Actually, `modalOverlay` already has a class in my CSS with all the required styles, just had to change 2 properties using JS to create the overlay effect. For some reason, CSS `opacity` property wasn't working on the class, hence I had to resort to this option 😅